Task2/TrainingManagementSystem/Training.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace TrainingManagementSystem
+{
+    internal class Training : TrainingComponent
+    {
+        List<TrainingComponent> trainingsAndExercises = new List<TrainingComponent>();
+
+        public Training()
+        {
+            trainingsAndExercises = new List<TrainingComponent>();
+        }
+
+        public List<TrainingComponent> TrainingsAndExercises => trainingsAndExercises;
+
+        public bool Add(TrainingComponent trainingComponent)
+        {
```

Comment:
1. If you use Contains you need to override Equals
2. What do you check by trainingComponent.GetType() != GetType()? Not to add Training to training? Use extra call definition to solve this ussie (look to previous lecture with my explanations)
3. Maybe TrainingsAndExercises.Contains(trainingComponent) != true if you try not to add the existing component?